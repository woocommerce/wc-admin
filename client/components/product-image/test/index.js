/** @format */
/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import ProductImage from '../';

describe( 'ProductImage', () => {
	test( 'should have correct alt text', () => {
		const image = <ProductImage alt="testing" />;
		expect( image.props.alt ).toBe( 'testing' );
	} );

	test( 'should have the correct width and height', () => {
		const image = <ProductImage width={ 30 } height={ 30 } />;
		expect( image.props.width ).toBe( 30 );
		expect( image.props.height ).toBe( 30 );
	} );

	test( 'should render a product image', () => {
		const product = {
			name: 'Test Product',
			images: [
				{
					src: 'https://i.cloudup.com/pt4DjwRB84-3000x3000.png',
				},
			],
		};
		const card = shallow( <ProductImage product={ product } /> );
		expect( card ).toMatchSnapshot();
	} );

	test( 'should render a placeholder image if no product images are found', () => {
		global.wcSettings.wcAssetUrl = 'https://woocommerce.com/wp-content/plugins/woocommerce/assets/';
		const product = {
			name: 'Test Product',
		};
		const card = shallow( <ProductImage product={ product } /> );
		expect( card ).toMatchSnapshot();
	} );
} );
