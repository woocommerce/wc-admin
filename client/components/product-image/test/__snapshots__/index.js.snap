// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should render a placeholder image if no product images are found 1`] = `
<img
  alt=""
  className="woocommerce-product-image is-placeholder"
  height={60}
  src="https://woocommerce.com/wp-content/plugins/woocommerce/assets/images/placeholder.png"
  width={60}
/>
`;

exports[`ProductImage should render a product image 1`] = `
<img
  alt=""
  className="woocommerce-product-image"
  height={60}
  src="https://i.cloudup.com/pt4DjwRB84-3000x3000.png"
  width={60}
/>
`;
